src/Stratis.Bitcoin/Controllers/Models/TransactionModel.cs

```diff
@@ -73,10 +73,16 @@ public TransactionVerboseModel(Transaction trx, Network network, ChainedHeader b
                 this.TxId = trx.GetHash().ToString();
                 this.Hash = trx.HasWitness ? trx.GetWitHash().ToString() : trx.GetHash().ToString();
                 this.Size = trx.GetSerializedSize();
-                this.VSize = trx.HasWitness ? trx.GetVirtualSize() : trx.GetSerializedSize();
+                this.VSize = trx.HasWitness ? trx.GetVirtualSize(network.Consensus.Options.WitnessScaleFactor) : trx.GetSerializedSize();
                 this.Version = trx.Version;
                 this.LockTime = trx.LockTime;
 
+                // size = (weight + WITNESS_SCALE_FACTOR - 1) / WITNESS_SCALE_FACTOR;
+                // hence, weight = size * WITNESS_SCALE_FACTOR - (WITNESS_SCALE_FACTOR - 1) (only subtract if has witness).
+                this.Weight = this.VSize * network.Consensus.Options.WitnessScaleFactor;
+                if (trx.HasWitness)
```

Comment:
Is this part definitely correct? The presence or absence of witness data is taken into account in `GetVirtualSize`. It seems to me that it would be simpler to unconditionally use `GetVirtualSize` to compute `VSize`. Then weight should obey the formula `weight = VSize * WITNESS_SCALE_FACTOR - (WITNESS_SCALE_FACTOR - 1)` without having to care if the transaction has witness data.

Reply:
Upon inspection, it is not needed. I removed the check